developers/tutorials/ipfs-filecoin/using-nftstorage.md

```diff
@@ -0,0 +1,451 @@
+             //Upload NFT to IPFS & Filecoin
+             const metadata = await nftStorage.store({
+                 name: 'Harmony NFT collection',
+                 description: 'This is a Harmony NFT collenction stored on IPFS & Filecoin.',
+                 image: file
+             });
+             return metadata;
+     
+         } catch (error) {
+             setErrorMessage("Could not save NFT to NFT.Storage - Aborted minting.");
+             console.log(error);
+         }
+     }
+     ```
+   
+     If NFT asset is uploaded successfully, NFT.Storage will do the following two things:
+
+     + Store NFT image on IPFS & Filecoin.
+     + Update NFT metadata with IPFS Link to your NFT image, then store NFT metadata on IPFS & Filecoin.
+   
+4. **Mint a new NFT on Harmony**
+
+   After getting the Metadata on IPFS, we can invoke the NFT smart contract on Harmony to mint a new NFT using NFT metadata on IPFS as tokenURI.
+
+   In this step, we will need to use `ethers` to interact with Metamask, so we first install this dependency.
+
+   ```shell
+   npm install ethers
+   ```
+
+   In order to invoke the NFT contract we have deployed to Harmony Testnet, we need to grab the contract ABI from the manifest JSON file which was generated via smart contract compilation.
+
+   ```shell
+   //run this command under harmony-nft folder
+   cp ../build/contracts/HarmonyNFT.json src/abi/
+   ```
+
+   Then we are ready to write code to send MintNFT transaction in `MintNFT.js` . First import ethers and smart contract manifest file,  and aslo create a `nftContractAddress` for the deployed NFT contract. 
+
+   ```javascript
+   import HarmonyNFT from "../abi/HarmonyNFT.json";
+   import { ethers } from 'ethers';
+   
+   const nftContractAddress = '0x494543afa8445025B9dBc7a39EdC7Fc624cA46F6';
+   ```
+
+   Then we can load the deployed NFT contract and invoke mintItem method.
+
+   ```javascript
+   //Add this code to load deployed NFT contract and invoke mintItem method.
+   const sendTxToHarmony = async(metadata) =>{
+       try {
+           const provider = new ethers.providers.Web3Provider(window.ethereum);
+           const connectedContract = new ethers.Contract(
+               nftContractAddress,
+               HarmonyNFT.abi,
+               provider.getSigner()
+           );
+           const mintNFTTx = await connectedContract.mintItem(metadata.url);
+           console.log(mintNFTTx);
+       } catch (error) {
+           setErrorMessage("Failed to send tx to Harmony.");
+           console.log(error);
+       }
+   }
+   ```
+
+5. **Preview the new minted NFT**
+
+   By now, we have all the codes to upload NFT data, send mint item transaction to Harmony. Let's add some code to preview our fresh minted NFT, including the NFT image, NFT metadata on IPFS and mintNFT transaction on Harmony.
+
+   You can find the full [MintNFT.js file on Github](https://github.com/longfeiWan9/harmony-nft-demo/blob/master/src/components/MintNFT.js).
```

Comment:
This transition feels a little bit unexpected. Are we telling people to open this file again?

Reply:
It is actually some extra code added in `MintNFT.js`, so I will  recommend developers to check the full file to see if they are writing the code correctly.